test/common/event/dispatcher_impl_test.cc

```diff
@@ -85,6 +85,7 @@ class DispatcherImplTest : public testing::Test {
     dispatcher_thread_->join();
   }
 
+  NiceMock<Stats::MockStore> scope_;
```

Comment:
Can you add something like `// Used by InitializeStats to ensure lives past the destructor and dispatcher exit.`?